test/e2e/serial/tests/workload_unschedulable.go

```diff
@@ -229,4 +231,140 @@ var _ = Describe("[serial][disruptive][scheduler] numaresources workload unsched
 			}
 		})
 	})
+
+	// other than the other tests, here we expect all the worker nodes (including none-bm hosts) to be padded
+	Context("with zero suitable nodes", func() {
+		var numOfnrtCandidates int
+		BeforeEach(func() {
```

Comment:
I'm not sold about new tests which have exactly one `It` block but a `BeforeEach` block. I think this is backwards: we should extract common code once we have it, not adding it proactively: it's too hard to predict the future and commit into a direction (any direction).
Anyway, I'm not blocking about this either.